lstests/core/helper.py

```diff
@@ -0,0 +1,148 @@
+    return time.strftime(fmt, convert(now))
+
+
+def make_tokens(amount: int, ndigits=4, symbol="SYS"):
+    """
+    Doctest
+    -------
+    >>> print(make_tokens(37.5e6))
+    37500000.0000 SYS
+    >>> print(make_tokens(1, 3, 'ABC'))
+    1.000 ABC
+    """
+    return "{{:.{}f}} {{}}".format(ndigits).format(amount, symbol)
+
+# --------------- string ----------------------------------------------------------------------------------------------
+
+def plural(word, count, suffix="s"):
+    if isinstance(word, tuple) or isinstance(word, list):
+        return word[0] if count <= 1 else word[1]
+    else:
+        return word if count <= 1 else  word + suffix
+
+
+def singular(word, count, suffix="s"):
+    if isinstance(word, tuple) or isinstance(word, list):
+        return word[0] if count <= 1 else word[1]
+    else:
+        return word  + suffix if count <= 1 else  word
+
+
+def pad(text, total, left=0, char=" ", sep="", textlen=None) -> str:
+    """textlen is a hint for visable length of text"""
+    textlen = vislen(text) if textlen is None else textlen
+    offset = len(text) - textlen
+    return (char * left + sep + text + sep).ljust(total + offset, char)
+
+
+def vislen(s: str):
+    """Return visible length of a string after removing color codes in it."""
+    return len(color.REGEX.sub("", s))
+
+
+def squeeze(s, maxlen, tail=0):
+    if len(s) <= maxlen:
+        return s
+    return s[:(maxlen-tail-2)] + ".." + (s[-tail:] if tail else "")
+
+# --------------- json ------------------------------------------------------------------------------------------------
+
+def abridge(data: typing.Union[dict, list], maxlen=79):
+    clone = copy.deepcopy(data)
+    trim(clone, maxlen=maxlen)
+    return clone
+
+
+def trim(data: typing.Union[dict, list], maxlen=79):
+    """
+    Summary
+    -------
+    A helper function for json formatting. This function recursively look
+    into a nested dict or list that could be converted to a json string,
+    and replace strings that are too long with three dots (...).
+    """
+    if isinstance(data, dict):
+        for key in data:
+            if isinstance(data[key], (list, dict)):
+                trim(data[key], maxlen=maxlen)
+            else:
+                if isinstance(data[key], str) and len(data[key]) > maxlen:
+                    data[key] = squeeze(data[key], maxlen=maxlen, tail=3)
+    elif isinstance(data, list):
+        for item in data:
+            if isinstance(item, (list, dict)):
+                trim(item, maxlen=maxlen)
+            else:
+                if isinstance(item, str) and len(item) > maxlen:
+                    item = squeeze(item, maxlen=maxlen, tail=3)
+
+# --------------- subprocess ------------------------------------------------------------------------------------------
+
+def get_pid_list_by_pattern(pattern: str) -> typing.List[int]:
+    out = subprocess.run(["pgrep", "-f", pattern], capture_output=True, text=True).stdout.splitlines()
```

Comment:
Thank you for the comment.

First, the assumption we made in the process of development is that we build with the availability of Python 3.7. If that assumption fails to hold, there are more places that we have to downgrade. For example, we rely on the `dataclasses` library introduced in Python 3.7. Python 3.7 has been released for one and a half years, so it might not be too aggressive to take advantage of the new features.

As to the choice of the names of the keyword arguments, being functionally equivalent, I guess it's more or less subjective. Personally I am more in favor of the newer ones in this case. Say `text` is shorter than `universal_newlines`. Again it's subjective.

Reply:
@b1aafulei The code will need to be modified to work with Python 3.6 or above. Python 3.6 is the highest version of Python that we are willing to require as a dependency of EOSIO at this time (we can support in a straightforward manner on all of our supported platforms accept Ubuntu 16.04; Ubuntu 16.04 will require building Python 3.7 from source in the CICD system).

Reply:
@arhag To honor the result of the discussion, I have adjusted the code here, and elsewhere as well, to work with Python 3.6. So here, as @jgiszczak suggested, I have made the changes accordingly. Thank you.